libraries/Microsoft.Bot.Builder/IPropertyContainer.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder
+{
+    public interface IPropertyContainer
+    {
+        /// <summary>
+        /// Get the property from the context
+        /// </summary>
+        /// <param name="turnContext"></param>
+        /// <returns></returns>
+        Task<T> GetPropertyAsync<T>(ITurnContext turnContext, string propertyName);
```

Comment:
I'm not seeing any value in these methods being `async` right now. State is always fully loaded at the point where this API kicks in and all of the implementation today is synchronous. The cost of the `Task` allocations and the complexity introduced by making these APIs `async` doesn't seem worth it... unless there is a plan to do the kind of deferred state loading that I was proposing in my DCR somewhere further down the line?

Reply:
Yeah, that was the idea.

---
In reply to: [203170557](https://github.com/Microsoft/botbuilder-dotnet/pull/760#discussion_r203170557) [](ancestors = 203170557)